jetty-server/src/main/java/org/eclipse/jetty/server/HttpInput.java

```diff
@@ -150,10 +151,11 @@ public boolean consumeAll()
     {
         try (AutoLock lock = _contentProducer.lock())
         {
-            IOException failure = new IOException("Unconsumed content");
+            if (_unconsumedContentException == null)
+                _unconsumedContentException = new IOException("Unconsumed content");
             if (LOG.isDebugEnabled())
-                LOG.debug("consumeAll {}", this, failure);
-            boolean atEof = _contentProducer.consumeAll(failure);
+                LOG.debug("consumeAll {}", this);
+            boolean atEof = _contentProducer.consumeAll(_unconsumedContentException);
```

Comment:
Can't we move the exception inside `ContentProducer`?

Reply:
As discussed, I've done it.